telemetry_package/main.py

```diff
@@ -1,34 +1,224 @@
+        "fcc_bridge.safety": 1000
+    }
+
+    def __init__(self):
+        """
+        Constructor function of the TelemetryNode class.
+
+        This function firstly provides the necessary QoS setting for each subscription, responsible for sending data to the ground-station webapp.
+
+        It then builds the UNIX socket used for the communication with the ground-station frontend and makes the node listen for a connection. The system will raise a warning if the socket is already in use.
+
+        After building the UNIX socket, this function creates two subscription: /rosout and TopicNames.Heartbeat.
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+
+        super().__init__('telemetry_node')
+
+        # Default QoS for the GroundStation - Node
+        qos_profile = QoSProfile(
+            history=QoSHistoryPolicy.KEEP_LAST,
+            depth=15,
+            reliability=QoSReliabilityPolicy.BEST_EFFORT,
+            durability=QoSDurabilityPolicy.VOLATILE,
+            liveliness=QoSLivelinessPolicy.AUTOMATIC
+        )
+
+        # For tracking the timestamps between the last received messages for throttled log-message names
+        self._message_throttle_table = {}
+
+        # Assume second and only arg as explicit path to unix socket
+        if len(sys.argv) > 1:
+            self.chosen_socket_path = sys.argv[1]
+        else:
+            self.chosen_socket_path = DEFAULT_UNIX_SOCKET_PATH
+
+        # Check if the socket already exists
+        if os.path.exists(self.chosen_socket_path):
+            raise FileExistsError('Socket already exists.')
+
+        try:
+            # Create the socket on the given path and listen for one connection
+            self.server_sock = socket.socket(
+                socket.AF_UNIX, socket.SOCK_STREAM)
+            self.server_sock.bind(self.chosen_socket_path)
+            print(
+                f"Listening for connection on the UNIX-socket at {self.chosen_socket_path} ...")
+            self.server_sock.listen(1)
+
+            # Accept the connection
+            self.client_sock, _ = self.server_sock.accept()
+            print('Connection accepted. Creating subscriptions...')
+
+            # Subscribe to /rosout
+            self.rosout_subscription = self.create_subscription(
+                Log,
+                '/rosout',
+                self.__rosout_callback,
+                qos_profile=qos_profile)
+
+            self.rosout_subscription
+            print('Created subscription to "/rosout".')
+
+            # Subscribe to TopicNames.Heartbeat
+            self.heartbeat_subscription = self.create_subscription(
+                Heartbeat,
+                TopicNames.Heartbeat,
+                self.__heartbeat_callback,
+                qos_profile=qos_profile)
+
+            self.heartbeat_subscription
+            print('created subscription to "heartbeat".')
+
+        except Exception as e:

```

Comment:
So here you handle the exception but you behave like nothing happened (just closing the socket but the rest of the program continues). You should exit the program here.